packages/shorebird_cli/test/src/artifact_builder_test.dart

```diff
@@ -195,117 +258,159 @@ void main() {
                 .thenReturn(ExitCode.success.code);
           });
 
-          group('when flutter is installed', () {
-            setUp(() {
-              when(() => operatingSystemInterface.which('flutter'))
-                  .thenReturn('/path/to/flutter');
-            });
-
-            test('runs flutter pub get with system flutter', () async {
-              await runWithOverrides(() => builder.buildAppBundle());
-
-              verify(
-                () => shorebirdProcess.run(
-                  'flutter',
-                  ['--no-version-check', 'pub', 'get', '--offline'],
-                  runInShell: any(named: 'runInShell'),
-                  useVendedFlutter: false,
-                ),
-              ).called(1);
-            });
-          });
+          verifyCorrectFlutterPubGet(
+            () async => runWithOverrides(() => builder.buildAppBundle()),
+          );
 
-          group('when flutter is not installed', () {
+          group('when the build fails', () {
             setUp(() {
-              when(() => operatingSystemInterface.which('flutter'))
-                  .thenReturn(null);
+              when(() => buildProcessResult.exitCode)
+                  .thenReturn(ExitCode.software.code);
             });
 
-            test('does not attempt to run flutter pub get', () async {
-              await runWithOverrides(() => builder.buildAppBundle());
-
-              verifyNever(
-                () => shorebirdProcess.run(
-                  'flutter',
-                  ['--no-version-check', 'pub', 'get', '--offline'],
-                  runInShell: any(named: 'runInShell'),
-                  useVendedFlutter: false,
-                ),
-              );
-            });
+            verifyCorrectFlutterPubGet(
+              () async => expectLater(
```

Comment:
Do we need these to be async if there is no await inside them?

Reply:
Good catch, removed throughout.